src/merkle/store.rs

```diff
@@ -469,27 +620,109 @@ mod test {
             (i3, LEAVES4[i3 as usize], p3),
         ];
 
+        let mut store = MerkleStore::default();
         store
             .add_merkle_paths(paths.clone())
             .expect("the valid paths must work");
 
-        let set = MerklePathSet::new(3).with_paths(paths).unwrap();
+        let depth = 3;
+        let set = MerklePathSet::new(depth).with_paths(paths).unwrap();
 
+        // STORE LEAVES ARE CORRECT ==============================================================
+        // checks the leaves in the store corresponds to the expected values
         assert_eq!(
-            set.get_node(NodeIndex::new(3, 0)).unwrap(),
-            store.get_node(set.root(), NodeIndex::new(2, 0b00)).unwrap(),
+            store
+                .get_node(set.root(), NodeIndex::new(set.depth() - 1, 0))
```

Comment:
I think we should fix this inconsistency, comments?

Reply:
Is the issue here with `MerklePathSet`?

Reply:
It seems that `MerklePathSet` is 1-indexed, wereas the other structures are 0-indexed.

In this test `mtree.depth()` is 2, and `set.depth()` is 3 because of this `+1`:

https://github.com/0xPolygonMiden/crypto/blob/962a07292f468e872f7453f13d9df28f2a78291c/src/merkle/path_set.rs#L136

Reply:
Ah - maybe related to this https://github.com/0xPolygonMiden/crypto/issues/57?

Reply:
seem related